src/Jackalope/Node.php

```diff
@@ -580,14 +588,23 @@ public function setProperty($name, $value, $type = PropertyType::UNDEFINED, $val
         }
 
         if (is_null($value)) {
-            if (isset($this->properties[$name])) {
-                $this->properties[$name]->remove();
+            if (isset($this->propertyValues[$name])) {
+                $this->loadProperty($name)->remove();
             }
 
             return null;
         }
 
-        return $this->_setProperty($name, $value, $type, false);
+        $this->_setProperty($name, $value, $type, false);
+
+        if (isset($this->properties[$name])) {
+            $property = $this->properties[$name];
+            $property->setValue($value, $type ?: PropertyType::UNDEFINED);
+        } else {
+            $property = $this->loadProperty($name);
+        }
```

Comment:
I think I have to find a cleaner solution here... But for now I just try to get the tests with jackalope-jackrabbit running again, and clean up afterwards.